### ISSUE TITLE
META Issue: Communication problems on v10.x

### ISSUE THREAD

**Issue by AlCalzone** [AlCalzone/node-zwave-js#4994]
It looks like there were some changes in v10, which break communication with some devices. We're still in the process of figuring things out and need some information to do so.

### If devices don't work (fully or partially) after upgrading, _first re-interview them_.

If they still don't work, please **share the following information** (attached as files):
1. a node dump
2. a driver log (loglevel "**silly**") of a re-interview
3. a driver log (loglevel "**silly**") of the actions that don't work
4. What does not work?

Comments that just say "me too" or similar and don't provide information will be deleted.

**Comment by bckmrk** [comment]
Hi, I have many devices not working after this update and attached is log No. 2 and 3 requested above. Where do i create a node dump? Logs above is specifically one device, Node 012 in my setup. 

2. [zwave_js-silly-Node012_re-interview.log](https://github.com/zwave-js/node-zwave-js/files/9470218/zwave_js-silly-Node012_re-interview.log)
3. [zwave_js-silly-Node012_lightsOn_NotWorking.log](https://github.com/zwave-js/node-zwave-js/files/9470219/zwave_js-silly-Node012_lightsOn_NotWorking.log)


4. The status of my dimmers updates in HA when I press the switches on the wall, but nothing happens with the lamps if I change the state of the switches on my phone or laptop. The switches in HA does not change either.

All of my devices that have lost some funtionality is Fibaro branded, and it is mostly Dimmer 2, FGD212 Firmware 3.5. I have a few other Fibaro devices working strange also...

**Comment by AlCalzone** [comment]
Your log shows that the dimmer accepted the command and sent an update after roughly one second.
```
# Turn on command sent:
2022-09-01T13:23:32.106Z DRIVER » [Node 012] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      50
                                  └─[SecurityCCCommandEncapsulation]
                                    │ nonce id: 238
                                    └─[MultiChannelCCCommandEncapsulation]
                                      │ source:      0
                                      │ destination: 1
                                      └─[MultilevelSwitchCCSet]
                                          target value: 255
                                          duration:     default

...

# Value changed to 99
2022-09-01T13:23:33.266Z CNTRLR   [Node 012] [~] [Multilevel Switch] currentValue: 0 => 99          [Endpoint 1]
2022-09-01T13:23:33.271Z CNTRLR   [Node 012] Scheduled poll canceled because expected value was received
2022-09-01T13:23:33.273Z DRIVER « [Node 012] [REQ] [ApplicationCommand]
                                  └─[SecurityCCCommandEncapsulation]
                                    │ sequenced: false
                                    └─[MultiChannelCCCommandEncapsulation]
                                      │ source:      1
                                      │ destination: 1
                                      └─[MultilevelSwitchCCReport]
                                          current value: 99
```

(some power usage updates are there aswell).

Are you sure this is the correct log?

**Comment by mirkochip88** [comment]
Qubino Flush 1D relay, node dead and not responding. Interviewing fail.
[zwave_js.log](https://github.com/zwave-js/node-zwave-js/files/9473117/zwave_js.log)
[zwave_js-6e431c384e81d642f9ea95e419f54ab6-Flush 1D relay-fbf0b7ebeb27bbac5eec9a483e422e80.json.txt](https://github.com/zwave-js/node-zwave-js/files/9473119/zwave_js-6e431c384e81d642f9ea95e419f54ab6-Flush.1D.relay-fbf0b7ebeb27bbac5eec9a483e422e80.json.txt)

**Comment by AlCalzone** [comment]
> Qubino Flush 1D relay, node dead and not responding

This claims it supports Multi Channel CC but does not respond to it at all. Interview gets aborted because this is considered a critical interview part.
Because it's Qubino, try excluding without reset and then include again. Their devices sometimes behave strangely when you change a config parameter that changes endpoints, but then don't re-include like the manual says.

**Comment by bckmrk** [comment]
> Your log shows that the dimmer accepted the command and sent an update after roughly one second.
> 
> ```
> # Turn on command sent:
> 2022-09-01T13:23:32.106Z DRIVER » [Node 012] [REQ] [SendData]
>                                   │ transmit options: 0x25
>                                   │ callback id:      50
>                                   └─[SecurityCCCommandEncapsulation]
>                                     │ nonce id: 238
>                                     └─[MultiChannelCCCommandEncapsulation]
>                                       │ source:      0
>                                       │ destination: 1
>                                       └─[MultilevelSwitchCCSet]
>                                           target value: 255
>                                           duration:     default
> 
> ...
> 
> # Value changed to 99
> 2022-09-01T13:23:33.266Z CNTRLR   [Node 012] [~] [Multilevel Switch] currentValue: 0 => 99          [Endpoint 1]
> 2022-09-01T13:23:33.271Z CNTRLR   [Node 012] Scheduled poll canceled because expected value was received
> 2022-09-01T13:23:33.273Z DRIVER « [Node 012] [REQ] [ApplicationCommand]
>                                   └─[SecurityCCCommandEncapsulation]
>                                     │ sequenced: false
>                                     └─[MultiChannelCCCommandEncapsulation]
>                                       │ source:      1
>                                       │ destination: 1
>                                       └─[MultilevelSwitchCCReport]
>                                           current value: 99
> ```
> 
> (some power usage updates are there aswell).
> 
> Are you sure this is the correct log?

Yes I'm sure it's the correct log. However I'm not sure the lamp didn't respond as intended... I wasn't at home doing this log so couldn't check... will redo log's later today.

**Comment by bckmrk** [comment]
Hmm, it's a little bit awkward to confess this, but re-interviewing every node that failed after the update helped with the problem. Now they seem to work perfectly again...

**Comment by AlCalzone** [comment]
Thanks for the feedback though.

**Comment by mirkochip88** [comment]
> > Qubino Flush 1D relay, node dead and not responding
> 
> This claims it supports Multi Channel CC but does not respond to it at all. Interview gets aborted because this is considered a critical interview part. Because it's Qubino, try excluding without reset and then include again. Their devices sometimes behave strangely when you change a config parameter that changes endpoints, but then don't re-include like the manual says.

It works, thanks!

**Comment by ekzor** [comment]
Hi there, I'm experiencing an issue where my Zooz ZSE19 siren will no longer play sound clips ever since I updated to the latest version of zwavejs2mqtt. Sound clips for this device are played through a configuration parameter being set to an integer. The sounds won't play anymore and it looks like the config parameter is basically stuck; it won't change no matter what I do.

Here are my node debug info (is that a node dump?), as well as Silly-level logs of the interview and the command that doesn't work.
- [zwavejs2mqtt-node_dump.txt](https://github.com/zwave-js/node-zwave-js/files/9481072/zwavejs2mqtt-node_dump.txt)
- [zwave_js-interview.log](https://github.com/zwave-js/node-zwave-js/files/9481036/zwave_js-interview.log)
- [zwave_js-not_working.log](https://github.com/zwave-js/node-zwave-js/files/9481042/zwave_js-not_working.log)

Versions:
- zwavejs2mqtt 7.0.0
- zwavejs 10.0.3

Things I've tried:

1. Healing
2. Reinterviewing
3. Restarting my zwavejs2mqtt docker container
4. Excluding / Reincluding
5. Reinterviewing again
6. Restarting again


Other things to note:

- I can see in the node's properties that `112-0-3` (the sound effect config parameter) is supposedly writeable, so I don't know why this is happening.
- The actual security siren functionality (i.e. the loud piercing wailing noise) and temperature/humidity reporting works just fine, i.e. I can turn the siren on and off and see values for the other stuff, so the node _is_ responsive. It's just setting the config parameter that doesn't seem to want to work.
- I haven't changed any of my settings or security keys, and my other S2-included devices (including other parts of this one) are still functional after the update.
- zwavejs2mqtt reports that the api call is successful although it's clear from the zwavejs driver logs that it actually isn't, but that's a different problem altogether :P

Things I'm about to try:

- Excluding / Reincluding without security (although ideally I wouldn't have to do this...) -- edit: nope, didn't work

Appreciate any help or advice. Thank you!

**Comment by skynet-network** [comment]
I have mostly fibaro dimmer and sunricher dimmer 

Going back to a 0.1.4  backup and unplugg "me zwave" stick removed from vm and add to vm rebooted host machine then all my zwave router was available and responded

**Comment by bckmrk** [comment]
My happiness was short-lived... now almost all of my z-wave devices are not working. Some commands pass through, but it takes a long, long time...  

[zwave_js_ReInterview.log](https://github.com/zwave-js/node-zwave-js/files/9483081/zwave_js_ReInterview.log)
[zwave_js_OnOff.log](https://github.com/zwave-js/node-zwave-js/files/9483082/zwave_js_OnOff.log)
[zwave_js-nodedump.json.txt](https://github.com/zwave-js/node-zwave-js/files/9483083/zwave_js-nodedump.json.txt)


Tried to heal the network and reinterview nodes again without any change. Suggestions?

Rolled back to 9.6.2 backup and everything works again.

**Comment by gralin** [comment]
I'm having issues with Zooz Open Close XS Sensor (ZSE41). I was doing the same test using latest HA addon 0.1.68 (zwave-js v10) and 0.1.65 (last zwave-js v9). In both tests I started with factory reset of the sensor and then tried to add it. The interview finished when using v9 but not in case of v10.

[zwave_js-37b187458f723b1a394110dcea8ea050-Node 20-ab82391c377dceca05b7f0bc778af64a.json.txt](https://github.com/zwave-js/node-zwave-js/files/9483425/zwave_js-37b187458f723b1a394110dcea8ea050-Node.20-ab82391c377dceca05b7f0bc778af64a.json.txt)
[log.txt](https://github.com/zwave-js/node-zwave-js/files/9483426/log.txt)

**Comment by epower53** [comment]
I'm struggling with a Zooz Zen16 configured as a garage door opener. Both relays 1 and 2 are configured for garage door mode (parameters 16-112-0-2 and 16-112-0-3 set to a value of 3). Working just fine before upgrading to ZwaveJS2mqtt version 7.0.0. Now, when I send a command to open or close either garage door, I hear a quick click from the relays, but it's too fast for the opener to register it as a physical button press and the doors do not open / close. Looking at the relay states in the ZwaveJS2mqtt GUI and watching topics using MQTT Explorer, it looks as though there's some serious funkiness going on. 

Manually changing the individual relay states from true to false or vice versa shows up in the targetValue and currentValue subtopics for the appropriate relays, however no action was actually taken by the device, aside from the quick click I hear. However, if I toggle the master device state for the whole Zen16 from false to true, both relays 1 and 2 fire simultaneously, for the correct momentary contact duration, and the garage doors are toggled. I'd prefer to have control over each door individually, as I did with the 6.X.X branch of ZwaveJS2mqtt and 9.X.X branch of Z-wave JS.

Current versions are ZwaveJS2mqtt 7.0.0, Z-wave JS 10.0.3. Here are the logs
[interview log.log](https://github.com/zwave-js/node-zwave-js/files/9483806/interview.log.log)
[driver level log.log](https://github.com/zwave-js/node-zwave-js/files/9483807/driver.level.log.log)

Edit: downgraded my install to ZwaveJS2Mqtt 6.16.0 and Z-wave JS to 9.6.2, and everything is back to normal. Will wait until the 10.X code branch has more bugs ironed out before trying it again.

**Comment by AlCalzone** [comment]
@ekzor the ZSE19 doesn't seem to accept supervised commands correctly. The v10 version started using those extensively to increase reliability through S2. I've notified Zooz and will disable Supervision for that device as a short term fix.

@zwave-js-bot add compat flag to `0x027a:0x000c:0x0003:0.0`
```jsonc
commandClasses: {
  remove: {
    // Supervised commands always result in a status of "Fail"
    "0x6c": {
      endpoints: "*"
    }
  }
}
```

**Comment by AlCalzone** [comment]
@bckmrk Your stick is hanging and does not respond to commands at all. It might need to be re-plugged.

> Error during node interview: Timeout while waiting for an ACK from the controller (ZW0200)
> 2022-09-03T17:21:48.947Z CNTRLR   Failed to execute controller command after 2/3 attempts. Scheduling next try i
                                  n 1100 ms.

**Comment by AlCalzone** [comment]
@gralin that log is not on loglevel "debug". Please make another one.

**Comment by gralin** [comment]
@AlCalzone I'm sorry, I have selected `silly` level in the Z-Wave JS addon configuration in HA but now when you pointed this out I noticed that after restarting, the log level of the addon reverted to `info`. Now when I selected `debug` instead of `silly` the setting survived the restart so I will perform the tests again and submit the proper logs. Is is enough when I copy the logs from the Log tab of the addon page? The start of the log seems to be truncated but I don't know where to get the complete ones.

**Comment by AlCalzone** [comment]
@gralin This works: https://www.home-assistant.io/integrations/zwave_js#how-to-access-the-z-wave-logs

**Comment by bckmrk** [comment]
> 

Is that usually needed after an update? I have never done that before. Will test it.

**Comment by bckmrk** [comment]
> @bckmrk Your stick is hanging and does not respond to commands at all. It might need to be re-plugged.
> 
> > Error during node interview: Timeout while waiting for an ACK from the controller (ZW0200)
> > 2022-09-03T17:21:48.947Z CNTRLR   Failed to execute controller command after 2/3 attempts. Scheduling next try i
> > n 1100 ms.

Is that usually needed after an update? Never done that before. Will test it. 

Everything worked well after I restored a backup, 0.1.65 (9.6.2). Did not re-plug anything.

**Comment by AlCalzone** [comment]
> Is that usually needed after an update?

It isn't usually, but it happens sometimes. Maybe the stick was still blocked by something else.

**Comment by AlCalzone** [comment]
@epower53 I'm seeing lots of connectivity issues in your log. 48 duplicate incoming commands during the interview, RSSI values close to -90 dBm (which isn't much more than the background noise), several instances where routing suddenly uses 3 hops instead of 0, timed out responses.

This hints at a bad connection between the node and the controller, in both directions. I suggest you check out and follow this:
- https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues?id=general-troubleshooting
- https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=testing-the-connection-strength (in zwavejs2mqtt on the network map after clicking a node)

The previous version probably works because all the information is already there, but the issue here is primarily in the real world.

**Comment by epower53** [comment]
Sorry to disagree with you @AlCalzone, but I'm going to have to. This device has been installed and working for over 2 years on both Hubitat and then Z-Wave JS 9.X.X. It's unsurprising the direct connection statistics are poor - there's a great deal of distance and a significant amount of wood, plaster, and concrete between the device and my controller. Usually this device runs through at least one relay, sometimes two, after my mesh has settled in. I know you can only base an opinion on what you see in the logs, so I'm providing this extra info so you understand that this device NEVER misbehaved until I tried 10.0.3. My guess is that it's another S2 issue. It's paired S2 unauthenticated, and you mentioned that the ZSE19 was not playing nice with supervision packets. It's something in the code (either Zooz's or Z-wave JS's).

**Comment by AlCalzone** [comment]
Can you share an interview log from 9.x then, so I can compare where the differences are?

Anyways, those are definitely not caused by Z-Wave JS:
```
2022-09-04T03:39:44.761Z DRIVER   Dropping message with invalid payload
2022-09-04T03:39:44.763Z DRIVER « [Node 016] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -75 dBm
                                  └─[Security2CCMessageEncapsulation] [INVALID]
                                      error: Duplicate command
2022-09-04T03:39:44.763Z SERIAL » [ACK]                                                                   (0x06)
2022-09-04T03:39:44.789Z SERIAL « 0x011900a8000110109f03a500d8111e1224b7dcba01ca6eee00a7aa            (27 bytes)
2022-09-04T03:39:44.791Z DRIVER   Dropping message with invalid payload
2022-09-04T03:39:44.792Z DRIVER « [Node 016] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -89 dBm
                                  └─[Security2CCMessageEncapsulation] [INVALID]
                                      error: Duplicate command
2022-09-04T03:39:44.793Z SERIAL » [ACK]                                                                   (0x06)
2022-09-04T03:39:44.853Z SERIAL « 0x011900a8000110109f03a500d8111e1224b7dcba01ca6eee00a8a5            (27 bytes)
2022-09-04T03:39:44.855Z DRIVER   Dropping message with invalid payload
2022-09-04T03:39:44.856Z DRIVER « [Node 016] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -88 dBm
                                  └─[Security2CCMessageEncapsulation] [INVALID]
                                      error: Duplicate command
```
and are 99.99% of the time a result of ACK frames that don't reach the device. And this is a problem when the device is busy repeating its recent commands instead of replying to the current one.

**Comment by epower53** [comment]
Happy to... here's an interview just completed using 9.6.2

[zwavejs_2022-09-05.log](https://github.com/zwave-js/node-zwave-js/files/9491115/zwavejs_2022-09-05.log)

**Comment by AlCalzone** [comment]
The connectivity issues are also there on 9.x, so timeouts because of them are more like a game of chance. If they happen at the wrong time you'll see isses after re-interviewing on 9.x too.

However the issue you're experiencing is a different one. The Zen16 does simply not respond when asked which commands each endpoint supports securely. 9.x (incorrectly) didn't do anything with this info anyways and simply used the information from the root device. This means this worked out most of the time by accident.

10.x assumes that the endpoint's commands are not secure instead - which is probably not the best choice if the device is secure. I can work around that, but ultimately it's the device's fault.

**Comment by epower53** [comment]
Zooz has been good offering support from an end customer's perspective  - hopefully they'll be willing to dig in and fix some of the issues the 10.X code branch is exposing. Thanks for interpreting the logs for me... Z-wave JS has been great since I made the switch.

Edit: @labob246 is having the same experience with the Zen16. Corroboration is great... means I'm NOT crazy.

**Comment by labob246** [comment]
I am also having a Zooz Zen16 issue.
[interview.log](https://github.com/zwave-js/node-zwave-js/files/9491711/interview.log)
[Actions that dont work.txt](https://github.com/zwave-js/node-zwave-js/files/9491713/Actions.that.dont.work.txt)

I do not know how to do a node dump, but the node in question is 15 and all nodes are directly connected to the stick.
In the Actions that don't work.txt file, you will see that the device can be turned on and off but there is no individual control of relays 1, 2 or 3.  In the file I turn on the device (relay 0), try turning on relay 3, try turning off relay 3 and then turn off the device.  The device (relay 0) controls the on/off of 1 or 2 or 3 when they should be able to be turned on/off individually, which doesn't work.  It stopped working with this latest version:  
zwavejs2mqtt: 7.0.0
zwave-js: 10.0.3

**Comment by gralin** [comment]
@AlCalzone coming back to you with `debug` logs. Same test as before: starting with factory reset Zooz Open Close XS Sensor (ZSE41) with firmware 1.10 and trying to add it to HA (I was using it successfully with HA for some time now). After adding the device I close and open the door once.

**Test 1**
Z-Wave JS 0.1.64
Driver Version: 9.4.0
Server Version: 1.20.0
Result: ✔️ **Device added to HA**
[zwave_js_adding_device_suceeded.log](https://github.com/zwave-js/node-zwave-js/files/9491984/zwave_js_adding_device_suceeded.log)
[zwave_js-success_node_diagnostics.json.txt](https://github.com/zwave-js/node-zwave-js/files/9491985/zwave_js-success_node_diagnostics.json.txt)

**Test 2**
Z-Wave JS 0.1.69
Driver Version: 10.0.3
Server Version: 1.22.1
Result: ❌ **Failed to add device to HA**
[zwave_js_adding_device_failed.log](https://github.com/zwave-js/node-zwave-js/files/9491993/zwave_js_adding_device_failed.log)
[zwave_js-failed_node_diagnostics.json.txt](https://github.com/zwave-js/node-zwave-js/files/9491994/zwave_js-failed_node_diagnostics.json.txt)

I have registered with Zooz and received latest firmware 1.30. Should I try to upgrade and do the test again?

**Comment by AlCalzone** [comment]
@gralin I found something buggy in your log that aborts the interview. However this is triggered by the device behaving strangely, which **could** be the result of a low battery:
```
2022-09-05T20:44:59.029Z DRIVER « [Node 021] [REQ] [BridgeApplicationCommand]
                                  │ RSSI: -91 dBm
                                  └─[Security2CCMessageEncapsulation]
                                    │ sequence number: 35
                                    └─[BatteryCCReport]
                                        level:  0
                                        is low: true
```

**Comment by AlCalzone** [comment]
@labob246 yours is the same issue as epower's

**Comment by bckmrk** [comment]
> > Is that usually needed after an update?
> 
> It isn't usually, but it happens sometimes. Maybe the stick was still blocked by something else.

Updated to 0.1.69, re-interviewed all Z-wave devices, re-plugged the stick. Now it's working. A little bit time-consuming, but thanks for your work and assistance. Greatly appreciated. :-)

**Comment by gralin** [comment]
@AlCalzone thank you for analyzing my logs ❤️ 

>I found something buggy in your log that aborts the interview. However this is triggered by the device behaving strangely, which could be the result of a low battery

I saw that too and that's why... I replaced the battery with a new one before the tests! The old one was due to be replaced anyway, but maybe I was out of luck and the new battery was faulty which I didn't consider. Especially because I was able to add the device (with that battery) using old version of the driver, but not with new one.

Today I have updated Z-Wave JS addon in HA and performed the tests again using driver 10.0.4. This time I managed to add the device! 🎉 Here is the log:  [zwave_js.log](https://github.com/zwave-js/node-zwave-js/files/9500539/zwave_js.log) If I am not mistaken, I have replaced the battery again after this becuase I saw in HA that reported battery level is 0%.

Finally I have upgraded the firmware in ZSE41 to latest 1.30 and re-added the device. No problems in this case either. HEre is the log: [zwave_js (1).log](https://github.com/zwave-js/node-zwave-js/files/9500549/zwave_js.1.log)

**Comment by m4dm4rtig4n** [comment]
Hello, since update to 7.X.X, all device in zwavejs2mqtt as disappeared :/
While in the logs I have just passed information about the nodes
I have try re-interviewed all Z-wave devices, reboot compute, But the node list is still empty.

Any idea why ?

**Comment by AlCalzone** [comment]
@m4dm4rtig4n please read the first comment in this thread and supply the information I ask for there.

**Comment by m4dm4rtig4n** [comment]
EDIT : i have shutdown master host (proxmox), unplug usb stick, reboot and now it's good

Start log available here :
https://pastebin.com/ckqpbJTA

re-interview log : 
https://pastebin.com/a5PbPdQj

Silly log of one node : 
```
2022-09-06T22:23:47.759Z SERIAL « 0x010700139b00000f7f                                                 (9 bytes)
2022-09-06T22:23:47.759Z SERIAL » [ACK]                                                                   (0x06)
2022-09-06T22:23:47.760Z DRIVER « [REQ] [SendData]
                                    callback id:     155
                                    transmit status: OK
2022-09-06T22:23:47.849Z SERIAL « 0x0113000400230d600d02005e0201050019001900f1                        (21 bytes)
2022-09-06T22:23:47.849Z SERIAL » [ACK]                                                                   (0x06)
2022-09-06T22:23:47.849Z CNTRLR   [Node 035] [translateValueEvent: value updated]
                                    commandClass: Z-Wave Plus Info
                                    endpoint:     0
                                    property:     zwavePlusVersion
                                    propertyKey:  undefined
                                    internal:     true
                                    secret:       false
                                    event source: undefined
2022-09-06T22:23:47.849Z CNTRLR   [Node 035] [~] [Z-Wave Plus Info] zwavePlusVersion: 1  [Endpoint 0] [internal]
                                  => 1
2022-09-06T22:23:47.849Z CNTRLR   [Node 035] [translateValueEvent: value updated]
                                    commandClass: Z-Wave Plus Info
                                    endpoint:     0
                                    property:     nodeType
                                    propertyKey:  undefined
                                    internal:     true
                                    secret:       false
                                    event source: undefined
2022-09-06T22:23:47.849Z CNTRLR   [Node 035] [~] [Z-Wave Plus Info] nodeType: 0 => 0     [Endpoint 0] [internal]
2022-09-06T22:23:47.849Z CNTRLR   [Node 035] [translateValueEvent: value updated]
                                    commandClass: Z-Wave Plus Info
                                    endpoint:     0
                                    property:     roleType
                                    propertyKey:  undefined
                                    internal:     true
                                    secret:       false
                                    event source: undefined
2022-09-06T22:23:47.849Z CNTRLR   [Node 035] [~] [Z-Wave Plus Info] roleType: 5 => 5     [Endpoint 0] [internal]
2022-09-06T22:23:47.850Z CNTRLR   [Node 035] [translateValueEvent: value added]
                                    commandClass: Z-Wave Plus Info
                                    endpoint:     2
                                    property:     installerIcon
                                    propertyKey:  undefined
                                    internal:     true
                                    secret:       false
                                    event source: undefined
2022-09-06T22:23:47.850Z CNTRLR   [Node 035] [+] [Z-Wave Plus Info] installerIcon: 6400  [Endpoint 2] [internal]
2022-09-06T22:23:47.850Z CNTRLR   [Node 035] [translateValueEvent: value added]
                                    commandClass: Z-Wave Plus Info
                                    endpoint:     2
                                    property:     userIcon
                                    propertyKey:  undefined
                                    internal:     true
                                    secret:       false
                                    event source: undefined
2022-09-06T22:23:47.850Z CNTRLR   [Node 035] [+] [Z-Wave Plus Info] userIcon: 6400       [Endpoint 2] [internal]
2022-09-06T22:23:47.850Z DRIVER « [Node 035] [REQ] [ApplicationCommand]
                                  └─[MultiChannelCCCommandEncapsulation]
                                    │ source:      2
                                    │ destination: 0
                                    └─[ZWavePlusCCReport]
                                        version:      1
                                        node type:    Node
                                        role type:    AlwaysOnSlave
                                        icon (mgmt.): 0x1900
                                        icon (user):  0x1900
2022-09-06T22:23:47.851Z CNTRLR « [Node 035] received response for Z-Wave+ information:             [Endpoint 2]
                                  Z-Wave+ version: 1
                                  role type:       AlwaysOnSlave
                                  node type:       Node
                                  installer icon:  0x1900
                                  user icon:       0x1900
2022-09-06T22:23:47.851Z CNTRLR   [Node 035] [translateValueEvent: value added]
                                    commandClass: Z-Wave Plus Info
                                    endpoint:     2
                                    property:     interviewComplete
                                    propertyKey:  undefined
                                    internal:     true
                                    secret:       false
                                    event source: undefined
2022-09-06T22:23:47.851Z CNTRLR   [Node 035] [+] [Z-Wave Plus Info] interviewComplete: t [Endpoint 2] [internal]
                                  rue
2022-09-06T22:23:47.851Z SERIAL » 0x010e00131707600d0001590103259c7c                                  (16 bytes)
2022-09-06T22:23:47.852Z DRIVER » [Node 023] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      156
                                  └─[MultiChannelCCCommandEncapsulation]
                                    │ source:      0
                                    │ destination: 1
                                    └─[AssociationGroupInfoCCNameGet]
                                        group id: 3
2022-09-06T22:23:47.852Z CNTRLR   [Node 035] Endpoint 2 interview: Supervision                      [Endpoint 2]
2022-09-06T22:23:47.852Z CNTRLR   [Node 035] Endpoint 2 interview: Binary Switch                    [Endpoint 2]
2022-09-06T22:23:47.852Z CNTRLR   [Node 035] Interviewing Binary Switch...                          [Endpoint 2]
2022-09-06T22:23:47.852Z CNTRLR » [Node 035] querying Binary Switch state...                        [Endpoint 2]
2022-09-06T22:23:47.854Z SERIAL « [ACK]                                                                   (0x06)
2022-09-06T22:23:47.861Z SERIAL « 0x0104011301e8                                                       (6 bytes)
2022-09-06T22:23:47.861Z SERIAL » [ACK]                                                                   (0x06)
2022-09-06T22:23:47.861Z DRIVER « [RES] [SendData]
                                    was sent: true
```

I have found this error :
2022-09-06T21:54:36.280Z DRIVER   Could not check for config updates: Failed to download package information! (ZW0343)

EDIT : i have shutdown master host (proxmox), unplug usb stick, reboot and now it's good

**Comment by jcustenborder** [comment]
I found this via this thread. [Missing Door State entity when adding Ecolink DWZWAVE25 sensors](https://community.home-assistant.io/t/missing-door-state-entity-when-adding-ecolink-dwzwave25-sensors/456395)

I have a DWZWAVE25 that I added a while ago and a few that are new. The new nodes are missing the state entity. I did notice that there is a difference in the nodes. I have attached a good and a bad node for comparison

[bad.txt](https://github.com/zwave-js/node-zwave-js/files/9531847/bad.txt)
[good.txt](https://github.com/zwave-js/node-zwave-js/files/9531848/good.txt)
[zwavejs_current.log](https://github.com/zwave-js/node-zwave-js/files/9531849/zwavejs_current.log)

**Comment by gralin** [comment]
@jcustenborder I had exactly same effect but with different door sensor (Zooz). Did you try updating zwave-js to latest version and re-pairing those devices? It helped in my case eventually (I also needed to replace the battery but that was just a coincidence).

**Comment by AlCalzone** [comment]
> re-pairing those devices

is usually not necessary. Re-interviewing is far easier.

**Comment by AlCalzone** [comment]
@jcustenborder yours is a duplicate of https://github.com/zwave-js/node-zwave-js/issues/5027

**Comment by jcustenborder** [comment]
@gralin Sorry I forgot to add I'm running `zwavejs/zwavejs2mqtt:7.1.0` of the docker container.

**Comment by jcustenborder** [comment]
@AlCalzone  Thank you! Is there any information that would be helpful to add to #5027

**Comment by AlCalzone** [comment]
Nope, the bug is known, I just need to fix it.

**Comment by AlCalzone** [comment]
@epower53 Zooz have told me that the Zen16 bug is known and a firmware with a fix exists. You might want to reach out to them to get the update.

**Comment by epower53** [comment]
Thanks @AlCalzone. Is this a different f/w than the latest release? I upgraded to the latest release, 1.04, last month. This is the version I was on when the bad behavior exposed by Z-wave JS 10.0 showed up.

**Comment by AlCalzone** [comment]
Not sure actually, I guess you could just ask them and mention that the maintainer of Z-Wave JS told you about the update.

**Comment by epower53** [comment]
Will do, thanks

**Comment by epower53** [comment]
Zooz support provided me with a newer (beta?) firmware to try on my Zen16. I'll apply it sometime later this week and re-try Z-wave JS 10.0.X and report back.

**Comment by finity69x2** [comment]
Is there a fix for this forthcoming?

It seems that we can't run v9.x of zwavejs along with v2022.9.x of HA. 

I'm getting this for the zwavejs integration:

"Retrying setup: Invalid server version: Z-Wave JS Server version is incompatible: 1.20.0 a version is required that supports at least api schema 22"

I'm running the following zwavejs after upgrading HA:

![image](https://user-images.githubusercontent.com/32221243/190007919-c2de984f-64ee-4cc3-a054-09203fced5c6.png)

If I update to v10 of zwavejs some of my zooz devices don't work.

So an update of either one has poor results.

**Comment by dkmaker** [comment]
> Is there a fix for this forthcoming?
> 
> 
> It seems that we can't run v9.x of zwavejs along with v2022.9.x of HA. 
> 
> 
> So an update of either one has poor results.

No you need to wait until there is a fix :-(

**Comment by AlCalzone** [comment]
> If I update to v10 of zwavejs some of my zooz devices don't work.

Is there an open issue for your problem yet?

**Comment by finity69x2** [comment]
I thought this was the one related to this not working on V10. Or did I not read the issue title or posts here correctly?

I could create another one but I don't think it will help if it's the same as this issue.

I was just wondering if this issue was close to being fixed yet. I added the datapoint about not being able to update HA because of it as additional info.

**Comment by AlCalzone** [comment]
@finity69x2 see the very first post. This issue is meant to collect information about **what** does not work and **why**. Just saying "it doesn't work" is not helpful.

**Comment by dkmaker** [comment]
> It looks like there were some changes in v10, which break communication with some devices. We're still in the process of figuring things out and need some information to do so.
> 
> 
> 
> ### If devices don't work (fully or partially) after upgrading, _first re-interview them_.
> 
> 
> 
> If they still don't work, please **share the following information** (attached as files):
> 
> 1. a node dump
> 
> 2. a driver log (loglevel "**silly**") of a re-interview
> 
> 3. a driver log (loglevel "**silly**") of the actions that don't work
> 
> 4. What does not work?
> 
> 
> 
> Comments that just say "me too" or similar and don't provide information will be deleted.

Do you need more tests? I can if needed make the tests with my fibaro units.. If you still havnt found the issue?

**Comment by AlCalzone** [comment]
> If you still havnt found the issue?

Which one? There were several, some related, some unrelated. This thread is meant to bundle all this investigation into one, so I don't have to look at 10 duplicate issues every day.

I'm not sure what is and what isn't working for you, or if your issues (if you had/have any) were already fixed.

Anyways, not a lot has surfaced in the past week, so unless someone reports back here with something new, I'll close this thread soon. The remaining issues I identified here seem to be related to S2 polishing and have existed before v10.

**Comment by Danymyte** [comment]
With the upgrade to v10 my Aeon Labs Garage Door Controller ZW062 is not working anymore. Trying to open or close the garage door results in no action at all. Tried to reinterview but without sucesss so far.

Just struggling how to collect all the data needed for bug fixing, can I get some advice?

Thanks a lot!

**Comment by TheDK** [comment]
> Do you need more tests? I can if needed make the tests with my fibaro units.. If you still havnt found the issue?

I just checked the following devices against the current version with todays release of Z-Wave JS to MQTT:

- Fibaro Single / Double Switch
- Fibaro Dimmer 2
- Fibaro Roller Shutter 3
- Fibaro Relay
- Fibaro Multisensor (Motion Eye)
- Steinel Outdoor PIR

All seem to be working just fine.

**Comment by dkmaker** [comment]
> I just checked the following devices against the current version with todays release of Z-Wave JS to MQTT:
>
> * Fibaro Dimmer 2
>
> All seem to be working just fine.

Okay - i'm using the Z-Wave JS addon in Homeassistant - and that breaks - is the change that went into Z-Wave JS to MQTT specfic for that? If that works then its great ..

**Comment by AlCalzone** [comment]
@zenturacp the official addon was also bumped to contain the fix: https://github.com/home-assistant/addons/blob/master/zwave_js/CHANGELOG.md#0171
make sure to re-interview after upgrading.

**Comment by TheDK** [comment]
Just FYI: No interview necessary on my part, everything just kept working as ever (I checked switches, states, meter).

**Comment by blommanx** [comment]
Hello, when upgrading to zwavejs 0.1.71 some of my Fibaro devices stops working (I didn't check all devices, but two Fibaro devices were the obvious ones).
They report the wrong state in Home Assistant and they don't respond to commands.
Reinterview on 71 did not help, rolling back to 0.1.65 is only way to fix it (had to reinterview on .65 and restart for them to come back).
Attached is logs from the Double Relay device
[zwave_js-diag-Double Switch 2.json.txt](https://github.com/zwave-js/node-zwave-js/files/9601231/zwave_js-58cd637ccb995fa3e52c9b69be6d0056-Double.Switch.2-e9ffc8d72c4dbc32ecd061585eaa963b.json.txt)
[zwave_js-interview.log](https://github.com/zwave-js/node-zwave-js/files/9601236/zwave_js.log)

**Comment by AlCalzone** [comment]
@blommanx The communication does look okay from the driver side though:
```
# command gets sent:
2022-09-19T17:45:20.534Z DRIVER » [Node 009] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      210
                                  └─[SecurityCCCommandEncapsulation]
                                    │ nonce id: 158
                                    └─[MultiChannelCCCommandEncapsulation]
                                      │ source:      0
                                      │ destination: 1
                                      └─[BinarySwitchCCSet]
                                          target value: true

# succeeds, value gets updated optimistically:
2022-09-19T17:45:20.651Z CNTRLR   [Node 009] [~] [Binary Switch] currentValue: true => true         [Endpoint 1]
2022-09-19T17:45:20.656Z CNTRLR   [Node 009] [+] [Binary Switch] targetValue: true                  [Endpoint 1]

# Change gets confirmed:
2022-09-19T17:45:21.793Z DRIVER » [Node 009] [REQ] [SendData]
                                  │ transmit options: 0x25
                                  │ callback id:      212
                                  └─[SecurityCCCommandEncapsulation]
                                    │ nonce id: 16
                                    └─[MultiChannelCCCommandEncapsulation]
                                      │ source:      0
                                      │ destination: 1
                                      └─[BinarySwitchCCGet]
...
2022-09-19T17:45:22.135Z CNTRLR   [Node 009] [~] [Binary Switch] currentValue: true => true         [Endpoint 1]
2022-09-19T17:45:22.138Z DRIVER « [Node 009] [REQ] [ApplicationCommand]
                                  └─[SecurityCCCommandEncapsulation]
                                    │ sequenced: false
                                    └─[MultiChannelCCCommandEncapsulation]
                                      │ source:      1
                                      │ destination: 0
                                      └─[BinarySwitchCCReport]
                                          current value: true

# And the device updates the state on its own again afterwards
2022-09-19T17:45:22.765Z CNTRLR   [Node 009] [~] [Binary Switch] currentValue: true => true         [Endpoint 1]
2022-09-19T17:45:22.767Z DRIVER « [Node 009] [REQ] [ApplicationCommand]
                                  └─[SecurityCCCommandEncapsulation]
                                    │ sequenced: false
                                    └─[MultiChannelCCCommandEncapsulation]
                                      │ source:      1
                                      │ destination: 0
                                      └─[BinarySwitchCCReport]
                                          current value: true
```
Those lines with `CNTRLR` indicate events that get emitted to applications, so HA should be seeing this.

You might want to update HA to see if this is already fixed on latest, you're still on 2022.8.7.

**Comment by TheDK** [comment]
Have you tried restarting HA? I had hang-ups on the web sockets from time to time.

**Comment by blommanx** [comment]
Interesting¸ somehow the interviews are messing with the state updates. Interviewing one that works breaks it again.
I've now upgraded, reinterviewed and restarted a bunch of times and finally they are responding properly.
Oh well, just need to figure out which ones need reinterview I guess.

**Comment by AlCalzone** [comment]
Yeah, both the v9 and v10 interview query things differently. v9 actually did not query everything, but it also (incorrectly) didn't use some of the information it should, so this missing info wasn't noticed until the upgrade and v10 relying on it.

**Comment by dkmaker** [comment]
> Interesting¸ somehow the interviews are messing with the state updates. Interviewing one that works breaks it again.
> 
> I've now upgraded, reinterviewed and restarted a bunch of times and finally they are responding properly.
> 
> Oh well, just need to figure out which ones need reinterview I guess.
> 
> 

Did the same but - reinterview one unit breaks others -

And some works but power reporting is not working on some of my units - it's really random what I see 

One of the units I can now control but the brightness is not updating but I could at least turn it on/off.

But I have 10+ fibaro dimmer devices so the right combo of reinterview etc. is difficult..

Restored again to 0.65 all is great again.. :-/ bummer

**Comment by AlCalzone** [comment]
No logs.. :-/ bummer

**Comment by sebastian-steinmann** [comment]
I'm not sure how to do a node-dump, using the hass-integration (not the mqtt version)
[zwave_js-reintervu.log](https://github.com/zwave-js/node-zwave-js/files/9624754/zwave_js-reintervu.log)
But here is the intervju logs, it just comes up as dead.

I have other devices on the network that comes up dead too, but right now they all work for some reason.

**Comment by AlCalzone** [comment]
@sebastian-steinmann that is because the node isn't reachable:
> transmit status: NoAck

If this is an intermittent issue, you should take a look at your mesh health. In any case, this is unrelated to v10.
1. https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues?id=general-troubleshooting
2. https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=testing-the-connection-strength

**Comment by sebastian-steinmann** [comment]
> @sebastian-steinmann that is because the node isn't reachable:
> 
> > transmit status: NoAck
> 
> If this is an intermittent issue, you should take a look at your mesh health. In any case, this is unrelated to v10.
> 
> 1. https://zwave-js.github.io/node-zwave-js/#/troubleshooting/connectivity-issues?id=general-troubleshooting
> 2. https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health?id=testing-the-connection-strength

I have tried most now, and it still won't respond. Excluded and tried to include it again, but still no luck. It's right next to a couple of other dimmers that do work. Don't have a backup to verify that it's it's connected to a specific version. Will try to debug further.

**Comment by dkmaker** [comment]
> No logs.. :-/ bummer

I spent some hours on it today - all actually looked good.. Did a ton of tests on Home Assistant ended up with a combo that worked

1. Reinvite the unit
2. Wait till the says unit ready
3. Restart the ZWave-JS Addon

Both power reporting and state worked as expected

Now i did

1. All units Reinvite
2. Wait for a long period of time until all units (25 was ack. and said OKAY)
3. Tested state - nothing still worked in Home Assistant - well probably switch worked but no reporting of power etc.
4. Restart ZWave JS Addon

Then all units works as expected .. 

So - guess the issue is somewhere between the addon and homeassistant - thought i had tested this already - but probably not as structured as i did today.

**Comment by epower53** [comment]
@AlCalzone I finally got around to updating my Zooz Zen16 to the new (now in public release) firmware, 1.10. Updated my ZwaveJS container to 10.3, with UI version 8.2.1 and tried again. All the multi-endpoint craziness is gone! Then Zen16 works perfectly with the new codebase now that I have the right firmware on it. Not that you were worried, but one more thing you can cross off the list.

**Comment by kaistierl** [comment]
I'm having issues with all of my Logic ZDB5400 Wall Controllers. Since the update to 10.x the ColorSwitch CC does not work anymore. A reinterview does help, but only for some time (I don't know how long exactly but it's around one or two days I'd say...) or until I restart the driver. After a driver restart, the ColorSwitch CC never works, until I re-interview.

I'm currently on 10.3.0, using it with zwave-js-ui connected to MQTT.

I did the following and collected driver logs of each step:

1. Tried to send a ColorSwitch Command with no success - [1_setvalue_error_before_reinterview.log](https://github.com/zwave-js/node-zwave-js/files/9833647/1_setvalue_error_before_reinterview.log)
2. Did a reinterview of the node - [2_reinterview.log](https://github.com/zwave-js/node-zwave-js/files/9833648/2_reinterview.log)
3. Sent another (now successful) ColorSwitch Command - [3_setvalue_success_after_reinterview.log](https://github.com/zwave-js/node-zwave-js/files/9833649/3_setvalue_success_after_reinterview.log)
4. Did a node dump - [4_nodedump_working_after_reinterview.json.txt](https://github.com/zwave-js/node-zwave-js/files/9833661/4_nodedump_working_after_reinterview.json.txt)
5. Restarted the driver and sent a ColorSwitch Command with no success again - [5_setvalue_error_after_restart.log](https://github.com/zwave-js/node-zwave-js/files/9833662/5_setvalue_error_after_restart.log)
6. Did another node dump (after driver restart) - [6_nodedump_broken_again_after_restart.json.txt](https://github.com/zwave-js/node-zwave-js/files/9833663/6_nodedump_broken_again_after_restart.json.txt)

@AlCalzone I would be really happy if you could help me... 🙂

**Comment by kaistierl** [comment]
In the meantime, I have done some more things. 

- Had a look in the logs myself. One thing that I noticed is that in the state where no ColorSwitch commands can be sent, the SPAN Extension is part of the SendData calls. In these cases, the command does not work. After a re-interview, the SPAN extension vanishes for some time and commands are sent fine. I suppose this has something to do with S2 encryption, but here my knowledge ends.
- Reinterview with "Reset Security Classes" set - didn't change behaviour
- Complete Re-Inclusion of one of the affected devices - did not work properly with 10.3.0, the device was always added without security.

For now I downgraded to zwave-js 9.6.2 (z2m 6.16.0). In this version, inclusion with S2 and everything else works again. This is fine for me until someone here can help 🙂

**Comment by AlCalzone** [comment]
@kaistierl I've opened a separate issue for your problem
